src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,264 @@
+            # Retrieving the content of the document
+            response = requests.get(
+                document_data.get("inhoud"),
+                auth=(settings.DOCUMENT_API_USER, settings.DOCUMENT_API_PASSWORD),
+            )
+            response.raise_for_status()
+
+            documents.append((document_data.get("titel"), response.content))
+
+        return documents
+
+    def _get_signers_from_package(self, package: dict) -> List[dict]:
+        """
+        Retrieves all the roles from a ValidSign package and returns all the signers.
+        """
+        logger.debug(
+            f"Retrieving the roles from validSign package '{package.get('id')}'"
+        )
+
+        roles_url = (
+            f"{settings.VALIDSIGN_ROOT_URL}api/packages/{package.get('id')}/roles"
+        )
+        response = requests.get(roles_url, headers=self._auth_header)
+        response.raise_for_status()
+        roles = response.json().get("results")
+        # Not all the roles are signers (one of them is the account owner)
+        return [role for role in roles if role.get("type") == "SIGNER"]
+
+    def create_package(self) -> dict:
+        """
+        Creates a ValidSign package with the name specified by the process variables and adds the signers to it.
+        """
+        logger.debug("Creating ValidSign package")
+
+        variables = self.task.get_variables()
+        signers = self.format_signers(variables.get("signers"))
+        package_name = variables.get("package_name")
+
+        url = f"{settings.VALIDSIGN_ROOT_URL}api/packages"
+
+        body = {
+            "name": package_name,
+            "type": "PACKAGE",
+            "roles": signers,
+        }
+
+        response = requests.post(url, headers=self._auth_header, data=json.dumps(body))
+        response.raise_for_status()
+        package = response.json()
+
+        return package
+
+    def add_documents_to_package(self, package: dict) -> List[dict]:
+        """
+        Adds documents to the specified package and returns a list with the information about each document returned by
+        ValidSign.
+        """
+
+        logger.debug(f"Adding documents to ValidSign package '{package.get('id')}'")
+
+        documents = self._get_documents_from_api()
+
+        post_url = (
+            f"{settings.VALIDSIGN_ROOT_URL}api/packages/{package.get('id')}/documents"
+        )
+
+        # Multiple files can be added in one request by passing the following 'files' parameter
+        # to the request, but then not sure how to specify the filename yet...
+        # files = [("files[]", content) for name, content in documents]
+
+        attached_documents = []
+        for doc_name, doc_content in documents:
+            payload = {"payload": f'{{"name": "{doc_name}"}}'}
+            file = [("file", doc_content)]
+            response = requests.post(
+                post_url, headers=self._auth_header, files=file, data=payload
+            )
```

Comment:
I don't trust the input here - a `doc_name` with value ` fo"o` will break the JSON (add a test for this!)

```suggestion
            payload = {"name": doc_name}
            post_data = {"payload": json.dumps(payload)}
            file = [("file", doc_content)]
            response = requests.post(
                post_url, headers=self._auth_header, files=file, data=post_data
            )
```